index.js

```diff
@@ -24,7 +28,7 @@ function base(to, from) {
 
 			if (Array.isArray(val)) {
 				to[key] = val.slice();
-			} else if (isObj(val)) {
+			} else if (isObj(val) && !isFunc(val)) {
 				to[key] = base(to[key] || {}, val);
 			} else if (val !== undefined) {
 				to[key] = val;
```

Comment:
Yes, It totally makes sense. I realised that `Object.assign` doesn't avoid functions, it does actually merge their own properties. Probably my attention missed this edge case, because I usually don't do this, sorry.